`error  '...' not found in imported namespace '...'  import/namespace` when using `export * from file.jsx`

Consider this:

```
src
├── bar
│   ├── baz.jsx
│   ├── index.js
│   └── qux.jsx
└── foo.js
```

**baz.jsx**

```
import React from "react";

export function Baz1() {
  return (
    <>
      <span>Baz1</span>
    </>
  );
}

export function Baz2() {
  return (
    <>
      <span>Baz2</span>
    </>
  );
}
```
**qux.jsx**

```
import React from "react";

export function Qux1() {
  return (
    <>
      <span>Qux1</span>
    </>
  );
}

export function Qux2() {
  return (
    <>
      <span>Qux2</span>
    </>
  );
}
```
**index.js**

```
export * from "./baz.jsx";
export { Qux1, Qux2 } from "./qux.jsx";
```

**foo.js**

```
import * as bar from "./bar/index.js";

console.log(bar.Baz1);
console.log(bar.Baz2);
console.log(bar.Qux1);
console.log(bar.Qux2);

```

When I try to lint this folder eslint-plugin-import will report:

```
d:\mcve_eslint\src\foo.js
  3:17  error  'Baz1' not found in imported namespace 'bar'  import/namespace
  4:17  error  'Baz2' not found in imported namespace 'bar'  import/namespace

✖ 2 problems (2 errors, 0 warnings)
```

The error is coming from https://github.com/import-js/eslint-plugin-import/blob/4bfe644ec697d0a4fe0227e5ff09b869650eb35f/src/rules/namespace.js#L134-L140

Isn't this a bug? Running react apps when using statements such as `export * from "./your_file.jsx"` seems to
be valid code.

As you can see when trying to export explicitely (`export { Qux1, Qux2 } from "./qux.jsx";`) there are no issues.

You can find the mcve [here](https://github.com/import-js/eslint-plugin-import/files/9784518/mcve_eslint.zip)

--- a/tests/files/jsx/bar/baz.jsx
+++ b/tests/files/jsx/bar/baz.jsx
@@ -1,0 +1,16 @@
+
+export function Baz1() {
+  return (
+    <div>
+    </div>
+  );
+}
+
+// Fragment Syntax
+export function Baz2() {
+  return (
+    <div>
+      <span>Baz2</span>
+    </div>
+  );
+}

--- a/tests/files/jsx/bar/index.js
+++ b/tests/files/jsx/bar/index.js
@@ -1,0 +1,2 @@
+export * from "./baz.jsx";
+export { Qux1, Qux2 } from "./qux.jsx";

--- a/tests/files/jsx/bar/qux.jsx
+++ b/tests/files/jsx/bar/qux.jsx
@@ -1,0 +1,16 @@
+
+export function Qux1() {
+  return (
+    <div>
+      <p>Qux1</p>
+    </div>
+  );
+}
+
+export function Qux2() {
+  return (
+    <div>
+      <p>Qux1</p>
+    </div>
+  );;
+}

--- a/tests/files/jsx/re-export.js
+++ b/tests/files/jsx/re-export.js
@@ -1,0 +1,1 @@
+export * from './named.jsx'

--- a/tests/src/rules/namespace.js
+++ b/tests/src/rules/namespace.js
@@ -29,6 +29,35 @@ const valid = [
       ecmaVersion: 2015,
     },
   }),
+  // import re-exported jsx files, where jsx file exports a string
+  test({
+    code: `
+      import * as foo from "./jsx/re-export.js";
+      console.log(foo.jsxFoo);
+    `,
+    settings: {
+      'import/extensions': ['.js', '.jsx'],
+    },
+  }),
+  // import re-exported jsx files, where jsx files export functions that return html tags
+  test({
+    code: `
+      import * as foo from "./jsx/bar/index.js";
+      console.log(foo.Baz1);
+      console.log(foo.Baz2);
+      console.log(foo.Qux1);
+      console.log(foo.Qux2);
+    `,
+    settings: {
+      'import/extensions': ['.js', '.jsx'],
+    },
+    parserOptions: {
+      ecmaFeatures: {
+        jsx: true,
+      },
+    },
+  }),
+
   test({ code: "import * as foo from './common';" }),
 
   // destructuring namespaces

